src/test/java/com/github/discvrseq/walkers/VcfToLuceneIndexerIntegrationTest.java

```diff
@@ -90,9 +97,61 @@ public void doBasicTest2() throws Exception {
         args.addRaw("-AN");
         args.addRaw("SampleList");
 
+        return args;
+    }
+
+    @Test
+    public void doBasicTest2() throws Exception {
+        File luceneOutDir = new File(getTmpDir(), "luceneOutDir");
+        if (luceneOutDir.exists())
+        {
+            FileUtils.deleteDirectory(luceneOutDir);
+        }
+
+        ArgumentsBuilder args = getBaseArgs(luceneOutDir);
         runCommandLine(args);
 
         File[] outputs = luceneOutDir.listFiles();
         Assert.assertEquals(5, outputs.length);
+
+        validateLuceneIndex(luceneOutDir);
+    }
+
+    @Test
+    public void doBasicTestWithMultithreading() throws Exception {
+        File luceneOutDir = new File(getTmpDir(), "luceneOutDir2");
+        if (luceneOutDir.exists())
+        {
+            FileUtils.deleteDirectory(luceneOutDir);
+        }
+
+        ArgumentsBuilder args = getBaseArgs(luceneOutDir);
+        args.addRaw("--threads");
+        args.addRaw("2");
+        runCommandLine(args);
+
+        File[] outputs = luceneOutDir.listFiles();
+        Assert.assertEquals(5, outputs.length);
+
+        validateLuceneIndex(luceneOutDir);
+    }
+
+    private void validateLuceneIndex(File indexPath) throws IOException, ParseException {
+        try (Directory indexDirectory = FSDirectory.open(indexPath.toPath());
+             IndexReader indexReader = DirectoryReader.open(indexDirectory)
+        ) {
+            IndexSearcher indexSearcher  = new IndexSearcher(indexReader);
```

Comment:
If MultiFieldQueryParser is causing everything to be interpreted as a Term, it means numeric queries like start:<3000 are getting interpreted lexicographically rather than numerically. to solve this issue, we can build the query ourselves rather than use the pre-made parser, which would look like this (untested):

```
IndexSearcher indexSearcher  = new IndexSearcher(indexReader);

TermQuery contigQuery = new TermQuery(new Term("contig", "1"));
TopDocs topDocs = indexSearcher.search(contigQuery, 10);
Assert.assertEquals(topDocs.totalHits.value, 16L);

TermQuery purposeQuery = new TermQuery(new Term("PURPOSE", "diff_pos_same_ref_same_alt"));
topDocs = indexSearcher.search(purposeQuery, 10);
Assert.assertEquals(topDocs.totalHits.value, 1L);

NumericRangeQuery<Integer> startQuery = NumericRangeQuery.newIntRange("start", 0, 3000, true, false);
topDocs = indexSearcher.search(startQuery, 10);
Assert.assertEquals(topDocs.totalHits.value, 16L);

```